activerecord/lib/active_record/table.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Table
+    mattr_accessor :base
+
+    extend self
+
+    ##
+    # Contains class methods corresponding to a table's columns, lazily evaluated
+    # at call time. Returns instances of <tt>Arel::Attributes::Attribute</tt>.
```

Comment:
Does this mean we should also remove `nodoc` from `Arel::Attributes::Attribute`?

Reply:
Ultimately yes, but that naturally opens up a whole cascade of methods and classes that should in turn be documented.

That's the point and the goal that this change starts us towards, but if we're going to make progress without this stalling horribly, we're going to have to accept that as we start down this road, there's going to be a gradually shifting border of things that are being exposed but haven't yet themselves been cleaned up for publishing.

So... yes, but no. :smile:

Reply:
![](https://media.giphy.com/media/5t9wJjyHAOxvnxcPNk/source.gif)